protocol/messenger_handler.go

```diff
@@ -1420,6 +1431,18 @@ func (m *Messenger) HandleChatIdentity(state *ReceivedMessageState, ci protobuf.
 		return err
 	}
 
+	contact := state.CurrentMessageState.Contact
+
+	if err = ValidateDisplayName(ci.DisplayName, ci.Version); err != nil {
```

Comment:
I think this could be problematic, it's a breaking change and it enforces users to set a username before sending messages, I had a word with John and I think the plan for rolling out the feature are incompatible with this change, but I'll ping you directly,  just to ask a few questions.